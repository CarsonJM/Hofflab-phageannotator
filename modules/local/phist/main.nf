process PHIST {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/phist:1.0.0--py310hd03093a_0':
        'biocontainers/phist:1.0.0--py310hd03093a_0' }"

    input:
    tuple val(meta) , path(virus_fastas)
    path(bacteria_fastas)

    output:
    tuple val(meta), path("${prefix}_common_kmers.csv") , emit: common_kmers
    tuple val(meta), path("${prefix}_predictions.csv")  , emit: predictions
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    phist.py \\
        ${virus_fastas} \\
        ${bacteria_fastas} \\
        ${prefix}_common_kmers.csv \\
        ${prefix}_predictions.csv \\
        -t ${task.cpus} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phist: \$(phist -v 2>&1 | sed '1!d; s/^.*PHIST utility //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_common_kmers.csv
    touch ${prefix}_predictions.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phist: \$(phist -v 2>&1 | sed '1!d; s/^.*PHIST utility //')
    END_VERSIONS
    """
}
