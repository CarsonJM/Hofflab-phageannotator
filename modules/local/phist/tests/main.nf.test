nextflow_process {

    name "Test Process PHIST"
    script "../main.nf"
    process "PHIST"
    config "./nextflow.config"


    test("virus_fasta_dir + bacteria_fasta_dir") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        "$projectDir/assets/test-datasets/phist/phages"
                    ]
                )
                input[1] = Channel.of(
                    [
                        "$projectDir/assets/test-datasets/phist/bacteria"
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        "$projectDir/assets/test-datasets/phist/phages"
                    ]
                )
                input[1] = Channel.of(
                    [
                        "$projectDir/assets/test-datasets/phist/bacteria"
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
