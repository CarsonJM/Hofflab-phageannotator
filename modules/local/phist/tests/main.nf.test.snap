{
    "virus_fasta_dir + bacteria_fasta_dir": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_common_kmers.csv:md5,3d11091d4ceb56b62fec132e158d4ec6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_predictions.csv:md5,9387252f5fc8ebec43017e66d484f85a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,568871dd9f6581b071673114c9f409be"
                ],
                "common_kmers": [
                    [
                        {
                            "id": "test"
                        },
                        "test_common_kmers.csv:md5,3d11091d4ceb56b62fec132e158d4ec6"
                    ]
                ],
                "predictions": [
                    [
                        {
                            "id": "test"
                        },
                        "test_predictions.csv:md5,9387252f5fc8ebec43017e66d484f85a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,568871dd9f6581b071673114c9f409be"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-06T16:00:38.129937147"
    },
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_common_kmers.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_predictions.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,079a94a9ff84a2b609f51a4879993c4c"
                ],
                "common_kmers": [
                    [
                        {
                            "id": "test"
                        },
                        "test_common_kmers.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "predictions": [
                    [
                        {
                            "id": "test"
                        },
                        "test_predictions.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,079a94a9ff84a2b609f51a4879993c4c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-06T16:00:48.151776203"
    }
}